app/gui/src/project-view/composables/nodeExecution.ts

```diff
@@ -0,0 +1,43 @@
+import { NodeId } from '@/stores/graph'
+import { type ProjectStore } from '@/stores/project'
+import { computed, reactive, ref } from 'vue'
+import { ExecutionEnvironment } from 'ydoc-shared/languageServerTypes'
+import { ExternalId } from 'ydoc-shared/yjsModel'
+
+/** Allows to recompute certain expressions (usually nodes). */
+export function useNodeExecution(projectStore: ProjectStore) {
```

Comment:
This is instantiated by `GraphNode` and `RecordControl`, but looks like it should be shared state so that e.g. a node's recomputation status would be influenced by the execution environment's global recomputation status..

Reply:
Good catch, fixed